base/freestanding/femr/config/migrations/next_appt_search_parameter.py

```diff
@@ -0,0 +1,48 @@
+import requests
+import json
+import logging
+import os
+
+# Migration script to add SearchParameter for next appointment extension
+revision = '985f4e1e-29f5-4911-bc9c-2c774b685289'
+down_revision = 'None'
+
+# Configure logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+# Define the FHIR server base URL
+FHIR_SERVER_URL = os.getenv('FHIR_URL')
+
+# Headers for FHIR requests
+HEADERS = {
+    'Content-Type': 'application/fhir+json'
+}
+SP_ID = "NextAppointmentTimeSearchParameter"
+
+
+def upgrade():
+    # Insert new SearchParameter
+    sp_resource = {
+        "resourceType": "SearchParameter",
+        "id": SP_ID,
+        "name": "NextAppointmentTimeSearchParameter",
+        "status": "active",
+        "description": "Search by patient extension for next appointment",
+        "code": "date-time-of-next-appointment",
+        "base": [ "Patient" ],
+        "type": "date",
+        "expression": "Patient.extension('http://www.uwmedicine.org/time_of_next_appointment').valueDateTime"
+    }
+    response = requests.put(f'{FHIR_SERVER_URL}SearchParameter/{SP_ID}', headers=HEADERS, data=json.dumps(sp_resource))
+    if response.status_code == 200 or response.status_code == 201:
+        logging.info('SearchParameter created successfully.')
+    else:
+        logging.error(f'Failed to create SearchParameter: {response.status_code} {response.text}')
```

Comment:
I might be missing some nuance in the HTTP codes returned by Hapi, but I always like this catch-all
```suggestion
    response.raise_for_status()
    logging.info('SearchParameter created successfully.')
```

Reply:
agreed!  i had used the examples but prefer this approach.  fhir-migrations will get better with use.

Reply:
Funny, I just ran into such a Hapi nuance at https://github.com/uwcirg/fhir-attrib-setter/blob/main/fhir-attrib-setter.patient.extension.py#L93 , Hapi might respond w/ a 200 code even though it had an error and didn't write anything. json for such a resonse: 
```
{"resourceType": "OperationOutcome", "issue": [{"severity": "error", "code": "processing", "diagnostics": "HAPI-0450: Failed to parse request body as JSON resource. Error was: HAPI-1821: [element=\"valueDateTime\"] Invalid attribute value \"2044-10-11 09:57:27\": Invalid date/time format: \"2044-10-11 09:57:27\": Expected character 'T' at index 10 but found  "}]}
```